modules/metering-troubleshooting.adoc

```diff
@@ -10,10 +10,173 @@ A common issue with metering is pods failing to start. Pods might fail to start
+
+. Check the `reporting-operator` pod for an `OOMKilled` termination:
++
+[source,terminal]
+----
+$ oc describe pod/reporting-operator-84f7c9b7b6-fr697
+----
++
+.Example output
+[source,terminal]
+----
+Name:         reporting-operator-84f7c9b7b6-fr697
+Namespace:    operator-metering
+Priority:     0
+Node:         ip-10-xx-xx-xx.ap-southeast-1.compute.internal/10.xx.xx.xx
+...
+   Ports:          8080/TCP, 6060/TCP, 8082/TCP
+   Host Ports:     0/TCP, 0/TCP, 0/TCP
+   State:          Running
+     Started:      Thu, 03 Dec 2020 20:59:45 +1000
+   Last State:     Terminated
+     Reason:       OOMKilled <1>
+     Exit Code:    137
+     Started:      Thu, 03 Dec 2020 20:38:05 +1000
+     Finished:     Thu, 03 Dec 2020 20:59:43 +1000
+----
+<1> The `reporting-operator` pod was terminated due to OOM kill.
+
+[discrete]
+[id="metering-check-and-increase-memory-limits_{context}"]
+=== Increasing the reporting-operator pod memory limit
+
+If you are experiencing an increase in pod restarts and OOM kill events, you can check the current memory limit set for the `reporting-operator` pod. Increasing the memory limit allows the `reporting-operator` pod to update the report data sources. If necessary, increase the memory limit in your `MeteringConfig` resource by 25-50%.
+
+.Procedure
+
+. Check the current memory limits of the `reporting-operator` pod:
++
+[source,terminal]
+----
+$ oc describe pod reporting-operator-67d6f57c56-79mrt
+----
++
+.Example output
+[source,terminal]
+----
+Name:         reporting-operator-67d6f57c56-79mrt
+Namespace:    operator-metering
+Priority:     0
+...
+   Ports:          8080/TCP, 6060/TCP, 8082/TCP
+   Host Ports:     0/TCP, 0/TCP, 0/TCP
+   State:          Running
+     Started:      Tue, 08 Dec 2020 14:26:21 +1000
+   Ready:          True
+   Restart Count:  0
+   Limits:
+     cpu:     1
+     memory:  500Mi <1>
+   Requests:
+     cpu:     500m
+     memory:  250Mi
+   Environment:
+...
+----
+<1> The current memory limit for the `reporting-operator` pod.
+
+. Edit the `MeteringConfig` resource to update the memory limit:
++
+[source,terminal]
+----
+$ oc edit meteringconfig/operator-metering
+----
++
+.Example `MeteringConfig` resource
+[source,yaml]
+----
+  kind: MeteringConfig
+  metadata:
+  name: operator-metering
+   namespace: operator-metering
+  spec:
+   openshift-reporting:
+     spec:
+       awsBillingReportDataSource:
+         bucket: gmeteringbucket
+         enabled: true
+         region: ap-southeast-1
+       resources: <1>
+         limits:
+           cpu: 1
+           memory: 750Mi
+         requests:
+           cpu: 500m
+           memory: 500Mi
+  ...
+----
+<1> Add or increase `memory` limits within the `resources` field of the `MeteringConfig` resource.
++
+[NOTE]
+====
+If memory limits are double the recommended defaults and there continue to be numerous OOM killed events, this might indicate that a different issue is causing the reports to be in a pending state.
```

Comment:
This is also likely fine, but our default recommendations are almost purely speculative and haven't been revisited (or even properly scale tested) in a while to feel confident with documenting this behavior.

Reply:
@timflannagan Attempt to make this more generic than "double the recommended defaults":

> If there continue to be numerous OOM killed events after memory limits are increased, this might indicate that a different issue is causing the reports to be in a pending state.